code/pagetypes/Forum.php

```diff
@@ -819,7 +823,7 @@ function PostMessageForm($addMode = false, $post = false) {
 			new FormAction("doPostMessageForm", _t('Forum.REPLYFORMPOST', 'Post'))
 		);
 
-		$required = $addMode === true ? new RequiredFields("Title", "Content") : new RequiredFields("Content");
+		$required = $addMode == 'true' ? new RequiredFields("Title", "Content") : new RequiredFields("Content");
```

Comment:
This is still going to cause problems because now if someone submits a value of `true` or `1` or `'TRUE'` this will fail.

How about:

```php

$required = new RequiredFields('Content');

if ($addMode) {
    $required->addRequiredField('Title');
}

``